.github/workflows/build-test.yml

```diff
@@ -0,0 +1,71 @@
+name: Build, Lint, and Test
+
+on:
+  workflow_call:
+
+jobs:
+  prepare:
+    name: Prepare
+    runs-on: ubuntu-latest
+    steps:
+        # v4.1.1
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      - name: Use Node.js
+        # v4.0.0
+        uses: actions/setup-node@8f152de45cc393bb48ce5d89d36b731f54556e65
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - name: Install Yarn dependencies
+        run: yarn --immutable
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    needs:
+      - prepare
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - run: yarn --immutable --immutable-cache
+      - run: yarn build
+      - name: Store build artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: build
+          path: ./build/
+      - name: Require clean working directory
+        shell: bash
+        run: |
+          if ! git diff --exit-code; then
+            echo "Working tree dirty at end of job"
+            exit 1
+          fi
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    needs:
+      - prepare
+    steps:
+        # v4.1.1
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      - name: Use Node.js
+        # v4.0.0
+        uses: actions/setup-node@8f152de45cc393bb48ce5d89d36b731f54556e65
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - run: yarn --immutable --immutable-cache
```

Comment:
Nit: We stopped using `--immutable-cache` because it would intermittently fail to cache the dependencies properly. When that happens, it was a pain to have to manually restart the job. Easier to have it not fail in the first place in that circumstance.

Same for the other `--immtable-cache` flag used above here in the `build` step

Reply:
It is removed now

Reply:
It looks like the flag is still here

Reply:
I believe I changed to wrong one, now it should be good, thanks!

Reply:
Looks like it's missing the `--immutable` flag now :sweat_smile:

Reply:
Just to clarify, this thread was about the `--immutable-cache` flag, which is different from the `--immutable` flag.

`--immutable` should be used for any installations that are not expected to update the lockfile. You'd only omit that in local development, or in CI jobs that intentionally update dependencies (e..g Dependabot).

Whereas `--immutable-cache`  is for when you want the installation to _fail_ if the cache is missing. We do expect the cache to exist in this case, but we've found that it's unreliable in practice, so in practice it's better not to expect that.